src/lwc/geFormWidgetRowAllocation/geFormWidgetRowAllocation.js

```diff
@@ -0,0 +1,230 @@
+import GAU_FIELD from '@salesforce/schema/Allocation__c.General_Accounting_Unit__c';
+const ALLOCATION_AMOUNT = `${ALLOCATION_OBJECT.objectApiName}.${AMOUNT_FIELD.fieldApiName}`;
+const ALLOCATION_PERCENT = `${ALLOCATION_OBJECT.objectApiName}.${PERCENT_FIELD.fieldApiName}`;
+const ALLOCATION_GAU = `${ALLOCATION_OBJECT.objectApiName}.${GAU_FIELD.fieldApiName}`;
+
+export default class GeFormWidgetRowAllocation extends LightningElement {
+    @api rowIndex;
+    @api rowRecord;
+    @api fieldList;
+    @api disabled;
+    @api totalAmount;
+    @api remainingAmount;
+
+    CUSTOM_LABELS = GeLabelService.CUSTOM_LABELS;
+
+    @api
+    getValues() {
+        const fields = this.template.querySelectorAll('c-ge-form-field');
+        let widgetFieldAndValues = {};
+        if(fields !== null && typeof fields !== 'undefined') {
+            fields.forEach(field => {
+                const { fieldAndValue } = field;
+                // convert full field names to local field names for processing
+                Object.entries(fieldAndValue).forEach(([fieldName,fieldValue]) => {
+                    const localFieldName = this.getLocalFieldName(fieldName);
+                    if(isNotEmpty(fieldValue)) {
+                        // no need to populate fields with values that are blank / empty
+                        widgetFieldAndValues = {...widgetFieldAndValues, ...{[localFieldName]: fieldValue}};
+                    }
+                });
+            });
+        }
+        return widgetFieldAndValues;
+    }
+
+    /**
+     * Given a field, set a value on that ge-form-field by calling ge-form-field.setFieldValue
+     * @param field Field name to set the value for
+     * @param value Value ot set
+     */
+    @api
+    setFieldValue(field, value) {
+        const element = this.getFieldByName(field);
+        if(element) {
+            element.setValue(value);
+        }
+    }
+
+    /**
+     * If the total amount is changed, any GAU allocated by percent must be re-allocated to match the total value.
+     * Call this method from the parent component on every allocation.
+     */
+    @api
+    reallocateByPercent(totalDonation) {
+        const percent = this.rowRecord[ALLOCATION_PERCENT];
+        if(isNumeric(percent) && percent >= 0) {
+            const detail = {
+                targetFieldName: ALLOCATION_PERCENT,
+                value: percent
+            };
+            this.handleChange({ detail }, totalDonation);
+        }
+    }
+
+    @api
+    isValid() {
+        const gauField = this.getFieldByName(ALLOCATION_GAU);
+        return gauField.isValid();
+    }
+
+    /**
+     * Calculate what amount is X percent of the total donation amount, rounded down to two decimal places.
+     * @param percentValue
+     * @param total
+     * @return {Number}
+     */
+    calculateAmount(percentValue, total) {
+        const factor = +percentValue / 100;
+        const amountInCents = Math.round((total * factor) * 100);
```

Comment:
Are we dividing by 100 and multiplying by 100  in this line? Why do we need the factor variable.

Reply:
Updated this with comments.